ls1intum/Artemis
# [Bugfix] Check for Invalid Short Name in Team Import

### Checklist

### Motivation and Context


Team names with leading digits lead to internal server errors
Fixes: #2854 

### Description

We are now checking for an invalid short name during the import of a teams json file. 

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a programming exercise in team mode
4. Add a team via the import via file option
5. The file looks like this:
Invalid file with leading digit in teamName:
```
[{"teamName": "1hallohallo", "username": "artemis_test_user_105"}]
```

File with valid teamName:
```
[{"teamName": "ValidName", "username": "artemis_test_user_105"}]
```
6. Verify that the invalid file throws an error message and the team cannot be imported.


### Test Coverage




### Screenshots


### Invalid team json:
![image](https://user-images.githubusercontent.com/33484318/109662308-32dd1980-7b6b-11eb-9d9a-455625ffc470.png)

### Valid setup:
#### Valid file:
![image](https://user-images.githubusercontent.com/33484318/109662525-70da3d80-7b6b-11eb-8778-54f196beb22a.png)
#### Resulting Dialog:
![image](https://user-images.githubusercontent.com/33484318/109662583-8485a400-7b6b-11eb-8123-5cf89a806306.png)
#### Imported team:
![image](https://user-images.githubusercontent.com/33484318/109662658-9404ed00-7b6b-11eb-9d1a-01ebb81a08c9.png)
